libs/feature/record/src/lib/ign-api-dl/ign-api-dl.component.ts

```diff
@@ -0,0 +1,249 @@
+import { BehaviorSubject, Observable, combineLatest, map, mergeMap } from 'rxjs'
+import { HttpClient } from '@angular/common/http'
+import { Choice } from '@geonetwork-ui/ui/inputs'
+import axios from 'axios'
+
+export interface Label {
+  label: string
+}
+
+export interface FormatProduit {
+  title: string
+  update: string
+  format: Array<TermBucket>
+  zone: Array<TermBucket>
+}
+
+export interface FormatSortieProduit {
+  label: string
+  value: string | number
+}
+export interface ListUrl {
+  url: string
+}
+
+export interface listChoice {
+  zone: Choice[]
+  format: Choice[]
+  editionDate: Choice[]
+  crs: Choice[]
+}
+
+export interface TermBucket {
+  term: string
+  label: string | number
+}
+
+export interface Field {
+  entry: Array<any>
+  link: any
+}
+
+@Component({
+  selector: 'gn-ui-ign-api-dl',
+  templateUrl: './ign-api-dl.component.html',
+  styleUrls: ['./ign-api-dl.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IgnApiDlComponent implements OnInit {
+  isOpen = false
+  collapsed = false
+  initialLimit = 50
+  apiBaseUrl: string
+  editionDate$ = new BehaviorSubject('')
+  zone$ = new BehaviorSubject('')
+  format$ = new BehaviorSubject('')
+  crs$ = new BehaviorSubject('')
+  page$ = new BehaviorSubject(1)
+  // a passer en config
+  url =
+    'https://data.geopf.fr/telechargement/capabilities?outputFormat=application/json'
+  choices: any
+  bucketPromisesZone: Choice[]
+  bucketPromisesFormat: Choice[]
+  bucketPromisesCrs: Choice[]
+
+  constructor(protected http: HttpClient) {}
+
+  @Input() set apiLink(value: DatasetServiceDistribution) {
+    this.apiBaseUrl = value ? value.url.href : undefined
+  }
+
+  ngOnInit(): void {
+    this.bucketPromisesZone = [{ value: '', label: 'ZONE' }]
+    this.bucketPromisesFormat = [{ value: '', label: 'FORMAT' }]
+    this.bucketPromisesCrs = [{ value: '', label: 'CRS' }]
+    this.getFields()
+  }
+
+  apiQueryUrl$ = combineLatest([
+    this.zone$,
+    this.format$,
+    this.editionDate$,
+    this.crs$,
+    this.page$,
+  ]).pipe(
+    map(([zone, format, editionDate, crs, page]) => {
+      let outputUrl
+      if (this.apiBaseUrl) {
+        const url = new URL(this.apiBaseUrl) // initialisation de l'url avec l'url de base
+        const params = {
+          zone: zone,
+          format: format,
+          editionDate: editionDate,
+          crs: crs,
+          page: page,
+        } // initialisation des paramètres de filtres
+        for (const [key, value] of Object.entries(params)) {
+          if (value && value !== 'null') {
+            url.searchParams.set(key, String(value))
+          } else {
+            url.searchParams.delete(key)
+          }
+        }
+        outputUrl = url.toString()
+      } else {
+        console.error('erreur apibaseUrl null')
+      }
+      return outputUrl
+    })
+    // startWith(() => this.apiBaseUrl)
```

Comment:
Is this still relevant? See also line 126 and 180.

Reply:
No, I removed it too.